extension/src/background/messageListener/freighterApiMessageListener.ts

```diff
@@ -92,21 +93,31 @@ export const freighterApiMessageListener = (
 
     const { _operations } = transaction;
 
-    const flaggedKeys = _operations.reduce(
-      (arr: Array<{}>, operation: { destination: string }) => {
-        const listing = accountData.find(
-          ({ address }: { address: string }) =>
-            address === operation.destination,
-        );
+    const flaggedKeys: FlaggedKeys = {};
```

Comment:
changing `FlaggedKeys` to an object prevents multiple entries that pertain to the same public key. Now, we will just update a public key's entry if we find more info about in on an iteration